internal/cluster/distribution/eks/eksworkflow/activity_update_addon.go

```diff
@@ -0,0 +1,187 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eksworkflow
+
+import (
+	"context"
+	"fmt"
+
+	"emperror.dev/errors"
+	"github.com/Masterminds/semver/v3"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/eks"
+	"go.uber.org/cadence/activity"
+
+	"github.com/banzaicloud/pipeline/internal/cluster/distribution/eks/eksprovider/workflow"
+	"github.com/banzaicloud/pipeline/internal/cluster/infrastructure/aws/awsworkflow"
+	"github.com/banzaicloud/pipeline/pkg/cadence/worker"
+)
+
+const UpdateAddonActivityName = "eks-update-addon"
+
+// UpdateAddonActivity responsible for updating an EKS addon
+// Addon is updated only in case it's already enable on cluster and there's a newer version available
+// for given Kubernetes version.
+type UpdateAddonActivity struct {
+	awsSessionFactory awsworkflow.AWSFactory
+	eksFactory        workflow.EKSAPIFactory
+}
+
+// UpdateAddonActivityInput holds data needed for updating an EKS addon
+type UpdateAddonActivityInput struct {
+	OrganizationID    uint
+	ProviderSecretID  string
+	Region            string
+	ClusterID         uint
+	ClusterName       string
+	KubernetesVersion string
+
+	AddonName string
+}
+
+// UpdateAddonActivityOutput holds the output data of the UpdateAddonActivityOutput
+type UpdateAddonActivityOutput struct {
+	UpdateID string
+}
+
+// NewUpdateAddonActivity instantiates a new EKS addon version update
+func NewUpdateAddonActivity(
+	awsSessionFactory awsworkflow.AWSFactory, eksFactory workflow.EKSAPIFactory,
+) *UpdateAddonActivity {
+	return &UpdateAddonActivity{
+		awsSessionFactory: awsSessionFactory,
+		eksFactory:        eksFactory,
+	}
+}
+
+// Register registers the activity in the worker.
+func (a UpdateAddonActivity) Register(worker worker.Registry) {
+	worker.RegisterActivityWithOptions(a.Execute, activity.RegisterOptions{Name: UpdateAddonActivityName})
+}
+
+func (a *UpdateAddonActivity) Execute(ctx context.Context, input UpdateAddonActivityInput) (*UpdateAddonActivityOutput, error) {
+	logger := activity.GetLogger(ctx).Sugar().With(
+		"organization", input.OrganizationID,
+		"cluster", input.ClusterName,
+		"region", input.Region,
+		"addonName", input.AddonName,
+	)
+
+	session, err := a.awsSessionFactory.New(input.OrganizationID, input.ProviderSecretID, input.Region)
+	if err = errors.WrapIf(err, "failed to create AWS session"); err != nil {
+		return nil, err
+	}
+
+	eksSvc := a.eksFactory.New(session)
+
+	describeAddonInput := &eks.DescribeAddonInput{
+		AddonName:   aws.String(input.AddonName),
+		ClusterName: aws.String(input.ClusterName),
+	}
+	addonOutput, err := eksSvc.DescribeAddon(describeAddonInput)
+	addonNotFoundErrMsg := fmt.Sprintf("No addon: %s found in cluster: %s", input.AddonName, input.ClusterName)
```

Comment:
**Recommended**: IMO the error message **SHOULD** be improved, the `: %s found in cluster` part is confusing in its current form after the `No addon` part, I propose a more straightforward phrasing.

```suggestion
	addonNotFoundErrMsg := fmt.Sprintf("No %s addon found in cluster: %s", input.AddonName, input.ClusterName)
```